### PR TITLE
MOSIP OIDP E-Signet National ID validation feature

### PR DESCRIPTION
**Pull request by naftis**
[MOSIP OIDP E-Signet National ID](https://docs.mosip.io/1.2.0/integrations/e-signet) authentication feature

This feature amends birth and death declaration forms, by adding a <kbd>Authenticate</kbd> -button in place of 'National ID' -text field if the integration is enabled. The user is then redirected to a configurable MOSIP url, where the user can complete verifying itself with a National ID. OpenCRVS then gets back a token, which is used to pre-fill the form with the user's details, and to confirm that the user's National ID has been verified.

### CODE REVIEW

File: packages/client/src/forms/configuration/default/birth.ts [hunk 1]
```diff
@@ -19,6 +19,31 @@ import {
   birthDocumentTypeFhirMapping
 } from '@client/forms/register/fieldMappings/birth/mutation/documents-mappings'
 
+const nidIntegrationConditionals = {
+  hideIfNidIntegrationEnabled: {
+    action: 'hide',
+    expression: `const nationalIdSystem =
+        offlineCountryConfig &&
+        offlineCountryConfig.systems.find(s => s.type === 'NATIONAL_ID');
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/default/birth.ts:27]: @naftis  These conditional needs to change to be exclusive to the MOSIP National ID type that is created in this PR: https://github.com/opencrvs/opencrvs-core/pull/4880

File: packages/client/src/forms/configuration/default/birth.ts [hunk 2]
```diff
@@ -19,6 +19,31 @@ import {
...
+        nationalIdSystem.settings.openIdProviderBaseUrl &&
+        nationalIdSystem.settings.openIdProviderClientId &&
+        nationalIdSystem.settings.openIdProviderClaims;
+    `
+  },
+  hideIfNidIntegrationDisabled: {
+    action: 'hide',
+    expression: `const nationalIdSystem = 
+      offlineCountryConfig &&
+      offlineCountryConfig.systems.find(s => s.type === 'NATIONAL_ID');
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/default/birth.ts:38]: as above

File: packages/client/src/forms/configuration/default/death.ts [hunk 3]
```diff
@@ -18,6 +18,31 @@ import {
   deathDocumentTypeFhirMapping
 } from '@client/forms/register/fieldMappings/death/mutation/documents-mappings'
 
+const nidIntegrationConditionals = {
+  hideIfNidIntegrationEnabled: {
+    action: 'hide',
+    expression: `const nationalIdSystem = 
+    offlineCountryConfig &&
+    offlineCountryConfig.systems.find(s => s.type === 'NATIONAL_ID');
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/default/death.ts:26]: as above

File: packages/client/src/forms/configuration/default/death.ts [hunk 4]
```diff
@@ -18,6 +18,31 @@ import {
...
+        nationalIdSystem.settings.openIdProviderBaseUrl &&
+        nationalIdSystem.settings.openIdProviderClientId &&
+        nationalIdSystem.settings.openIdProviderClaims;
+    `
+  },
+  hideIfNidIntegrationDisabled: {
+    action: 'hide',
+    expression: `const nationalIdSystem = 
+    offlineCountryConfig &&
+    offlineCountryConfig.systems.find(s => s.type === 'NATIONAL_ID');
```
Comment (euanmillar) [inline packages/client/src/forms/configuration/default/death.ts:37]: as above

File: packages/client/src/utils/nationalIdUtils.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ */
+import { getOfflineData } from '@client/offline/selectors'
+import { useSelector } from 'react-redux'
+
+export function useIsNationalIdVerificationEnabled() {
+  const offlineData = useSelector(getOfflineData)
+
+  return offlineData.systems.some(
+    (integration) =>
+      integration.type === 'NATIONAL_ID' &&
```
Comment (euanmillar) [inline packages/client/src/utils/nationalIdUtils.ts:20]: as above

File: packages/client/src/views/SysAdmin/Config/Systems/mutations.ts [hunk 6]
```diff
@@ -89,8 +95,10 @@ export const updateSystemPermissions = gql`
       status
       type
       settings {
-        event
-        permissions
+        webhook {
```
Comment (euanmillar) [inline packages/client/src/views/SysAdmin/Config/Systems/mutations.ts:98]: @naftis thanks for fixing this!

File: packages/client/src/components/form/FormFieldGenerator.tsx [hunk 7]
```diff
@@ -967,12 +988,33 @@ class FormSectionComponent extends React.Component<Props> {
...
+                          fields: fieldsWithValuesDefined
+                        }
+                      ]
+                    } as IFormSection
+
+                    const form = {
+                      sections: [section]
+                    } as IForm
+                  }
+                  //TODO: HANDLE FETCH FOR NID
```
[the remark this answers is not in the document]
Comment (naftis) [reply packages/client/src/components/form/FormFieldGenerator.tsx:1009]: https://github.com/opencrvs/opencrvs-core/pull/4933 This is being worked at here fyi

**Pull request merged**